### PR TITLE
[DoctrineBridge] fix case sensitivity issue in RememberMe\DoctrineTokenProvider

### LINKED ISSUE #21467: DoctrineTokenProvider issue with postgresql
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | yes
| Symfony version  | 3.2.1

There is an issue with DoctrineTokenProvider if using postgresql.

The problem is that by default postgresql is not case-sensitive unless you apply quotes to your arguments.
Thus creating `lastUsed` table as suggested might result in creating it all-lowercase. All is fine untill you hit:
```php
    return new PersistentToken($row['class'], $row['username'], $series, $row['value'], new \DateTime($row['lastUsed']));
```
 which will throw exception as there's no such field in table. Looks like changing the field to be exactly `lastUsed` might help, but for me exceptions start happening even earlier, since this time some other queries could not obtain this field properly. I ended up replacing `'lastUsed'` to be `'lastused'` in that particular line.

Therefore I wonder is there any better approach to fix this? 

As I'm afraid fixing this way might break this code for mysql or any other database.

### PR DESCRIPTION
**Pull request by PF4Public**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | ?
| Fixed tickets | #21467
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by PF4Public** [comment]
Hi @nicolas-grekas 
Looks like I did it, please check if I've done it corectly.

**Comment by nicolas-grekas** [comment]
We now need to figure out how not to break existing apps.

**Comment by PF4Public** [comment]
Should I rebase it against newest version, which allows BC breaks?

**Comment by nicolas-grekas** [comment]
We always strive for creating a continuous upgrade path so that there is never a hard BC break. This means e.g. triggering a deprecation notice when applicable, etc.

**Comment by fabpot** [comment]
How can we move forward here?

**Comment by PF4Public** [comment]
Since this change could in fact break existing installations, it probably needs to be done in two steps: with deprecation notice first.
Sadly, I simply do not know, how are such situations handled in Symfony.

**Review by nicolas-grekas — APPROVED** [review]
I pushed a new fix for this issue on your fork.
PR ready.

**Comment by PF4Public** [comment]
@nicolas-grekas , Thank you for fixing this. You've found a really clever way!

File: src/Symfony/Bridge/Doctrine/Security/RememberMe/DoctrineTokenProvider.php [hunk 1]
```diff
@@ -50,15 +50,16 @@ public function __construct(Connection $conn)
      */
     public function loadTokenBySeries($series)
     {
-        $sql = 'SELECT class, username, value, lastUsed'
+        // the alias for lastUsed works around case insensitivity in PostgreSQL
+        $sql = 'SELECT class, username, value, lastUsed as last_used'
```
Comment (fabpot) [inline src/Symfony/Bridge/Doctrine/Security/RememberMe/DoctrineTokenProvider.php]: `AS`?

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @PF4Public.

**Pull request merged**